runtime/lua/vim/fs.lua

```diff
@@ -227,7 +231,20 @@ end
 ---@return (string) Normalized path
 function M.normalize(path)
   vim.validate({ path = { path, 's' } })
-  return (path:gsub('^~/', vim.env.HOME .. '/'):gsub('%$([%w_]+)', vim.env):gsub('\\', '/'))
+  local p = (path:gsub('^~/', vim.env.HOME .. '/'):gsub('%$([%w_]+)', vim.env):gsub('\\', '/'))
+
+  -- Try to also convert the path to an absolute path with symlinks resolved, but if the path does
+  -- not yet exist simply return it as-is.
+  return M.realpath(p) or p
+end
+
+--- Convert a path to an absolute path, resolving symlinks.
+---
+---@param path (string) Path to convert to an absolute path
+---@return (string|nil) Absolute path, or nil if the file or directory does not
+---        exist.
+function M.realpath(path)
+  return vim.loop.fs_realpath(path)
```

Comment:
[quick comment on mobile]

Luv is already a part of our API and is documented via `:help luvref` so there is no need for a redundant wrapper here.

Reply:
In general we should strongly avoid redundant wrappers.  OTOH,

1. vim.loop is low-level
2. when we introduce "topic" modules like `vim.fs`, there is an expectation that they cover the topic, so all related functions can be discovered there.
3. We may enhance wrappers added here in the future. If plugins are all using low-level stuff, they won't get those improvements.

Reply:
>vim.loop is low-level

I don’t feel strongly about it, but this was the reason for introducing this. I have heard in the past that `vim.loop` is *caveat emptor* territory, so this is just a way to provide a “user facing” interface to this one particular API.

Otherwise I agree about redundant wrappers.

Reply:
> luv is not part of that supported space, it's low level, like FFI.

Err, that is news to me, and quite concerning since **many** plugins are using `vim.loop`. I've always understood it as being very supported.

Was it then a mistake to include the luv documentation in our runtime?

Reply:
> quite concerning since many plugins are using vim.loop

Nothing to be concerned about, but it is similar story as FFI: we exposed a low-level thing that we don't own. 

> Was it then a mistake to include the luv documentation in our runtime?

No, it's useful and used :)